hotel/site_sections/hotel_assignments.py

```diff
@@ -243,16 +243,19 @@ def gaylord(self, out, session):
 
     @csv_file
     def requested_hotel_info(self, out, session):
-        attendees_without_hotel_pin = session.query(Attendee).filter(
-                Attendee.requested_hotel_info == True,
-                Attendee.hotel_pin == None).all()
+        hotel_query = session.query(Attendee).filter(
+            Attendee.badge_status != c.INVALID_STATUS,
+            Attendee.requested_hotel_info == True,
+            Attendee.first_name != '')
+
+        attendees_without_hotel_pin = hotel_query.filter(Attendee.hotel_pin == None).all()
         if attendees_without_hotel_pin:
             for a in attendees_without_hotel_pin:
                 a.hotel_pin = _generate_hotel_pin()
             session.commit()
 
         out.writerow(['First Name', 'Last Name', 'E-mail Address', 'Password'])
-        for a in session.query(Attendee).filter_by(requested_hotel_info=True).order_by(Attendee.first_name, Attendee.last_name):
+        for a in sorted(hotel_query.all(), key=lambda a: a.legal_first_name + ' ' + a.legal_last_name):
```

Comment:
The `legal_first_name` and `legal_last_name` are big guesses on our part... is there anything about this export that needs that name to be split? In other words, can we just sort on `legal_name`?

Reply:
Lol, nope! That's a good suggestion!